src/NorthwindWebsite/Middleware/Handlers/Interfaces/IImageCachingHandler.cs

```diff
@@ -0,0 +1,14 @@
+﻿namespace NorthwindWebsite.Middleware.Handlers.Interfaces;
+
+public interface IImageCachingHandler
+{
+    bool IsContained(string imageIndex);
+
+    int GetNumberOfFilesInCachingFolder();
+
+    void CreateFolderIfDoesNotExists();
```

Comment:
Well, in previous comment I meant that you should split your method into 2 separate methods: one that checks that folder really exists, the second one is that only creates method. However, in second method you may leave validation of created folder to make sure your app has no access issues to file server system.